cl/alerts/send_alerts.py

```diff
@@ -104,46 +147,110 @@ def send_rt_alerts(response: Response, document_data: dict[str, Any]) -> None:
         if not alert_triggered:
             continue
 
-        alert_user: UserProfile.user = alert_triggered.user
-        not_donated_enough = (
-            alert_user.profile.total_donated_last_year
-            < settings.MIN_DONATION["rt_alerts"]
-        )
-        if not_donated_enough and alert_triggered.rate == Alert.REAL_TIME:
-            logger.info(
-                "User: %s has not donated enough for their "
-                "RT alerts to be sent.\n" % alert_user
-            )
-            continue
-        qd = QueryDict(alert_triggered.query.encode(), mutable=True)
+        # Schedule no RT Alerts.
+        scheduled_rates = [Alert.DAILY, Alert.WEEKLY, Alert.MONTHLY]
+        if alert_triggered.rate in scheduled_rates:
+            with transaction.atomic(savepoint=True):
+                try:
+                    (
+                        user_rate,
+                        created,
+                    ) = UserRateAlert.objects.select_for_update().get_or_create(
+                        user=alert_triggered.user, rate=alert_triggered.rate
+                    )
+                except IntegrityError:
+                    user_rate = UserRateAlert.objects.get(
+                        user=alert_triggered.user, rate=alert_triggered.rate
+                    )
+                parent_alert, created = ParentAlert.objects.get_or_create(
+                    alert=alert_triggered, user_rate=user_rate
+                )
+                ScheduledAlertHit.objects.create(
+                    parent_alert=parent_alert,
+                    document_content=document_content,
+                    highlighted_fields=hit.meta["highlight"].to_dict(),
+                )
 
-        # Set highlight if available in response.
-        if hasattr(hit.meta, "highlight"):
-            merge_highlights_into_result(
-                hit.meta["highlight"], document_data, ALERTS_HL_TAG
-            )
+        # Send RT Alerts
         if alert_triggered.rate == Alert.REAL_TIME:
+            alert_user: UserProfile.user = alert_triggered.user
+            not_donated_enough = (
+                alert_user.profile.total_donated_last_year
+                < settings.MIN_DONATION["rt_alerts"]
+            )
+            if not_donated_enough and alert_triggered.rate == Alert.REAL_TIME:
```

Comment:
I think we can remove the expression that checks the value of the `rate` attribute in this `if statement` because the `statement` on Line 175 already checks the value of that attribute.

Reply:
Yeah, you right, let's remove `alert_triggered.rate == Alert.REAL_TIME` from the `IF`, thanks.